src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,45 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+
+        try {
+            Rectangle rectangle = (Rectangle)obj;
+            if (color == null) {
+                if (rectangle.color != null) {
+                    return false;
+                }
+            }
+            if (obj.getClass().equals(Rectangle.class)) {
+                return ((color == null && rectangle.color == null)
+                        || rectangle.getWidth().equals(getWidth()))
+                        && ((width == null && rectangle.width == null)
+                        || rectangle.getLength().equals(getLength()))
+                        && ((length == null && rectangle.length == null)
+                        || rectangle.getColor().equals(getColor()));
+            }
+
+        } catch (ClassCastException e) {
+            throw new RuntimeException("obj can't be casted to Rectangle", e);
+        }
```

Comment:
but if cast after checking, i have no access to rectangle fields of Object, because it's not an Rectangle yet, and it has no field width, for example... i am confuced...

Reply:
I guess i have got: i must cast exactly after cheking object type. Done.